### PR TITLE
stage1: improve error for missing a number type on a runtime var

### PR DESCRIPTION
**Pull request by pfgithub**
The previous error was [really unclear](https://www.reddit.com/r/Zig/comments/jamfyt/feedback_from_a_new_user_of_zig/), suggesting you should make a variable const even when there is a runtime type the value could be casted to instead

```zig
var i = 10;
```
will now report `error: numbers modified at runtime must be given a size, such as i32 or u64. either specify a type or make this variable const` instead of `error: variable of type 'comptime_int' must be const or comptime`

```zig
var i = 10.5;
```
will now report `error: numbers modified at runtime must be given a size, such as f32 or f64. either specify a type or make this variable const` instead of `error: variable of type 'comptime_float' must be const or comptime`

### CODE REVIEW

**Comment by andrewrk** [comment]
Thanks for addressing this.

> such as i32 or u64

Hmm this is problematic - I don't want to confuse newbies into thinking these particular integer types are special. I can foresee people blindly using i32 or u64 without thinking it through. The original error message is valuable as well. Can you do this as a hint?

My counter proposal:

`error: variable of type 'comptime_int' must be const or comptime` (same)
`note: to modify 'i' at runtime the variable must be given a fixed-size number type`

**Comment by pfgithub** [comment]
I can't build zig locally and the CI is broken right now, so I'm not sure if this is right

**Comment by andrewrk** [comment]
No problem. I'm working on the CI. My macbook takes all day to compile LLVM

File: test/compile_errors.zig [hunk 1]
```diff
@@ -6912,7 +6912,9 @@ pub fn addCases(cases: *tests.CompileErrorContext) void {
         "tmp.zig:2:4: error: variable of type '*const comptime_int' must be const or comptime",
         "tmp.zig:5:4: error: variable of type '(undefined)' must be const or comptime",
         "tmp.zig:8:4: error: variable of type 'comptime_int' must be const or comptime",
+        "tmp.zig:8:4: note: to modify this variable at runtime, it must be given an explicit fixed-size number type",
         "tmp.zig:11:4: error: variable of type 'comptime_float' must be const or comptime",
+        "tmp.zig:11:4: note: to modify this variable at runtime, it must be given an explicit fixed-size float type",
```
Comment (andrewrk) [inline test/compile_errors.zig:6917]: Looks like a typo:

```suggestion
        "tmp.zig:11:4: note: to modify this variable at runtime, it must be given an explicit fixed-size number type",
```

**Pull request merged**